galley/pkg/runtime/distributor.go

```diff
@@ -0,0 +1,50 @@
+//  Copyright 2018 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package runtime
+
+import (
+	sn "istio.io/istio/galley/pkg/mcp/snapshot"
+)
+
+// Distributor interface allows processor to distribute snapshots of configuration.
+type Distributor interface {
+	SetSnapshot(name string, snapshot sn.Snapshot)
+
+	ClearSnapshot(name string)
+}
+
+// InMemoryDistributor is an in-memory distributor implementation.
+type InMemoryDistributor struct {
```

Comment:
Does this need locking? Or is this intentionally not thread safe assuming the caller will only ever invoke from a single context. If the latter, a comment about this restriction would be good.